packages/server/src/automations/tests/scenarios/scenarios.spec.ts

```diff
@@ -49,151 +49,188 @@ describe("Automation Scenarios", () => {
         },
       })
     })
-  })
-
-  it("should trigger an automation which querys the database", async () => {
-    const table = await config.createTable()
-    const row = {
-      name: "Test Row",
-      description: "original description",
-      tableId: table._id,
-    }
-    await config.createRow(row)
-    await config.createRow(row)
-    const builder = createAutomationBuilder({
-      name: "Test Row Save and Create",
-    })
 
-    const results = await builder
-      .appAction({ fields: {} })
-      .queryRows({
-        tableId: table._id!,
+    it("should trigger an automation which querys the database", async () => {
+      const table = await config.createTable()
+      const row = {
+        name: "Test Row",
+        description: "original description",
+        tableId: table._id,
+      }
+      await config.createRow(row)
+      await config.createRow(row)
+      const builder = createAutomationBuilder({
+        name: "Test Row Save and Create",
       })
-      .run()
 
-    expect(results.steps).toHaveLength(1)
-    expect(results.steps[0].outputs.rows).toHaveLength(2)
-  })
+      const results = await builder
+        .appAction({ fields: {} })
+        .queryRows({
```

Comment:
we should build a test scenario that loops over the results of a query rows - not for this PR, but just got me thinking as I see it! Feels like one of the most common use cases for it.

Reply:
I've got another branch with a few tests on it and this is one of them, definitely an important one to get covered.